adafruit_wm8960.py

```diff
@@ -21,48 +25,1587 @@
+AVDD=3.3V
+'''
+
+WM8960_VSEL_INCREASED_BIAS_CURRENT = 1
+WM8960_VSEL_LOWEST_BIAS_CURRENT = 3
+
+WM8960_REGISTER_DEFAULTS = [
+    0x0097, # R0 (0x00)
+    0x0097, # R1 (0x01)
+    0x0000, # R2 (0x02)
+    0x0000, # R3 (0x03)
+    0x0000, # R4 (0x04)
+    0x0008, # F5 (0x05)
+    0x0000, # R6 (0x06)
+    0x000A, # R7 (0x07)
+    0x01C0, # R8 (0x08)
+    0x0000, # R9 (0x09)
+    0x00FF, # R10 (0x0a)
+    0x00FF, # R11 (0x0b)
+    0x0000, # R12 (0x0C) RESERVED
+    0x0000, # R13 (0x0D) RESERVED
+    0x0000, # R14 (0x0E) RESERVED
+    0x0000, # R15 (0x0F) RESERVED
+    0x0000, # R16 (0x10)
+    0x007B, # R17 (0x11)
+    0x0100, # R18 (0x12)
+    0x0032, # R19 (0x13)
+    0x0000, # R20 (0x14)
+    0x00C3, # R21 (0x15)
+    0x00C3, # R22 (0x16)
+    0x01C0, # R23 (0x17)
+    0x0000, # R24 (0x18)
+    0x0000, # R25 (0x19)
+    0x0000, # R26 (0x1A)
+    0x0000, # R27 (0x1B)
+    0x0000, # R28 (0x1C)
+    0x0000, # R29 (0x1D)
+    0x0000, # R30 (0x1E) RESERVED
+    0x0000, # R31 (0x1F) RESERVED
+    0x0100, # R32 (0x20)
+    0x0100, # R33 (0x21)
+    0x0050, # R34 (0x22)
+    0x0000, # R35 (0x23) RESERVED
+    0x0000, # R36 (0x24) RESERVED
+    0x0050, # R37 (0x25)
+    0x0000, # R38 (0x26)
+    0x0000, # R39 (0x27)
+    0x0000, # R40 (0x28)
+    0x0000, # R41 (0x29)
+    0x0040, # R42 (0x2A)
+    0x0000, # R43 (0x2B)
+    0x0000, # R44 (0x2C)
+    0x0050, # R45 (0x2D)
+    0x0050, # R46 (0x2E)
+    0x0000, # R47 (0x2F)
+    0x0002, # R48 (0x30)
+    0x0037, # R49 (0x31)
+    0x0000, # R50 (0x32) RESERVED
+    0x0080, # R51 (0x33)
+    0x0008, # R52 (0x34)
+    0x0031, # R53 (0x35)
+    0x0026, # R54 (0x36)
+    0x00e9, # R55 (0x37)
+]
 
 class WM8960:
-    def __init__(self, i2c_bus, address=0x1a):
-        self.i2c_device = i2c_device.I2CDevice(i2c_bus, address)
+    def __init__(self, i2c_bus:I2C, address:int = WM8960_ADDR):
+        self.i2c_device = I2CDevice(i2c_bus, address)
         self._buf = bytearray(2)
 
-    def _write_reg(self, reg, value):
+        '''
+        The WM8960 does not support I2C reads
+        This means we must keep a local copy of all the register values
+        We will instantiate with default values by copying from WM8960_REGISTER_DEFAULTS during reset()
+        As we write to the device, we will also make sure to update our local copy as well, stored here in this array.
+        Each register is 9-bits
+        They are in order from R0-R55, and we even keep blank spots for the "reserved" registers. This way we can use the register address macro defines above to easiy access each local copy of each register.
+        Example: self._registerLocalCopy[WM8960_REG_LEFT_INPUT_VOLUME]
+        '''
+        self._registerLocalCopy = [0x0000 for i in range(len(WM8960_REGISTER_DEFAULTS))]
+        self.reset()
+
+        # General setup
+        self.enableVREF()
+        self.enableVMID()
+
+    def isConnected(self) -> bool:
+        # TODO: Check I2C or I2CDevice
+        return True
```

Comment:
I don't think you need this because I2CDevice will check that the device exists on the bus.
```suggestion
```